drivers/counter/counter_mcux_lpc_rtc.c

```diff
@@ -0,0 +1,251 @@
+{
+	const struct counter_config_info *info = dev->config;
+	const struct mcux_lpc_rtc_config *config =
+		CONTAINER_OF(info, struct mcux_lpc_rtc_config, info);
+
+	return RTC_GetStatusFlags(config->base) & RTC_CTRL_ALARM1HZ_MASK;
+}
+
+static uint32_t mcux_lpc_rtc_get_top_value(const struct device *dev)
+{
+	const struct counter_config_info *info = dev->config;
+
+	return info->max_top_value;
+}
+
+static void mcux_lpc_rtc_isr(const struct device *dev)
+{
+	const struct counter_config_info *info = dev->config;
+	const struct mcux_lpc_rtc_config *config =
+		CONTAINER_OF(info, struct mcux_lpc_rtc_config, info);
+	struct mcux_lpc_rtc_data *data = dev->data;
+	counter_alarm_callback_t cb;
+	uint32_t current = mcux_lpc_rtc_read(dev);
+
+
+	LOG_DBG("Current time is %d ticks", current);
+
+	if ((RTC_GetStatusFlags(config->base) & RTC_CTRL_ALARM1HZ_MASK) &&
+	    (data->alarm_callback)) {
+		cb = data->alarm_callback;
+		data->alarm_callback = NULL;
+		cb(dev, 0, current, data->alarm_user_data);
+	}
+
+	if (data->top_callback) {
+		data->top_callback(dev, data->top_user_data);
+	}
+
+	/*
+	 * Clear any conditions to ack the IRQ
+	 *
+	 * callback may have already reset the alarm flag if a new
+	 * alarm value was programmed to the TAR
+	 */
+	RTC_StopTimer(config->base);
+	if (RTC_GetStatusFlags(config->base) & RTC_CTRL_ALARM1HZ_MASK) {
+		RTC_ClearStatusFlags(config->base, kRTC_AlarmFlag);
+	}
+	RTC_StartTimer(config->base);
+}
+
+static int mcux_lpc_rtc_init(const struct device *dev)
+{
+	const struct counter_config_info *info = dev->config;
+	const struct mcux_lpc_rtc_config *config =
+		CONTAINER_OF(info, struct mcux_lpc_rtc_config, info);
+
+	RTC_Init(config->base);
+
+	config->irq_config_func(dev);
+
+	return 0;
+}
+
+static const struct counter_driver_api mcux_rtc_driver_api = {
+	.start = mcux_lpc_rtc_start,
+	.stop = mcux_lpc_rtc_stop,
+	.get_value = mcux_lpc_rtc_get_value,
+	.set_alarm = mcux_lpc_rtc_set_alarm,
+	.cancel_alarm = mcux_lpc_rtc_cancel_alarm,
+	.set_top_value = mcux_lpc_rtc_set_top_value,
+	.get_pending_int = mcux_lpc_rtc_get_pending_int,
+	.get_top_value = mcux_lpc_rtc_get_top_value,
+};
+
+static struct mcux_lpc_rtc_data mcux_lpc_rtc_data_0;
+
+static void mcux_lpc_rtc_irq_config_0(const struct device *dev);
+
+static struct mcux_lpc_rtc_config mcux_lpc_rtc_config_0 = {
+	.base = (RTC_Type *)DT_INST_REG_ADDR(0),
+	.irq_config_func = mcux_lpc_rtc_irq_config_0,
+	.info = {
+		.max_top_value = UINT32_MAX,
+		.freq = 1,
+		.flags = COUNTER_CONFIG_INFO_COUNT_UP,
+		.channels = 1,
+	},
+};
+
+DEVICE_DT_INST_DEFINE(0, &mcux_lpc_rtc_init, NULL,
+		    &mcux_lpc_rtc_data_0, &mcux_lpc_rtc_config_0.info,
+		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
+		    &mcux_rtc_driver_api);
+
+static void mcux_lpc_rtc_irq_config_0(const struct device *dev)
+{
+	IRQ_CONNECT(DT_INST_IRQN(0),
+		    DT_INST_IRQ(0, priority),
+		    mcux_lpc_rtc_isr, DEVICE_DT_INST_GET(0), 0);
+	irq_enable(DT_INST_IRQN(0));
+}
```

Comment:
Please update this to use `DT_INST_FOREACH_STATUS_OKAY`

Reply:
Done.